.github/workflows/ci-cd.yml

```
@@ -1,31 +1,62 @@
+---
 name: CI/CD
 
 on:
   push:
     branches:
-      - 'master'
-    tags: [ '[0-9]+.[0-9]+.[0-9]+' ]
+    - master
+    tags:
+    - '[0-9]+.[0-9]+.[0-9]+'
   pull_request:
     branches:
-      - 'master'
-      - '[0-9].[0-9]+.[0-9]+'
+    - master
+    - '[0-9].[0-9]+.[0-9]+'
 
 env:
   FORCE_COLOR: 1
 
 jobs:
+  build:
+    name: Build distribution 📦
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: 3.x
+    - name: Install build dependencies
+      run: python -m pip install build twine
+    - name: Build distribution 📦
+      run: python -m build
+    - name: Check distribution 📦
+      run: python -m twine check --strict dist/*
+    - name: Upload distribution 📦
+      uses: actions/upload-artifact@v4
+      with:
+        name: python-package-distributions
+        path: dist/
+
   test:
     name: Test Python ${{ matrix.python-version }}
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12', '3.13']
+        python-version:
+        - '3.8'
+        - '3.9'
+        - '3.10'
+        - '3.11'
+        - '3.12'
+        - '3.13'
```

it is possible to unquote these as follows:
```suggestion
        - 3.8
        - 3.9
        - >-
          3.10
        - 3.11
        - 3.12
        - 3.13
```